services/frontmatter_generator.py

```
@@ -2,503 +2,194 @@
-            "og:description": description,
-            "og:type": "article",
-            "og:url": f"{self.frontend_domain}/blog/{target_date}",
-            "og:image": best_image,
-            "og:site_name": "Daily Devlog"
-        }
-        
-        # Build enhanced BlogPosting schema with rich content
-        blog_posting_schema = {
+    """Generate clean frontmatter for blog posts."""
+
+    def __init__(self, author: str, base_url: str, media_domain: str, frontend_domain: str = None):
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)